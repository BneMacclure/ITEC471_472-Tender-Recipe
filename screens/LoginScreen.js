import React, { Component , useState} from "react";
import {
    StyleSheet,
    View,
    Image,
    ImageBackground,
    Text,
    TextInput, 
    TouchableOpacity,
    ActivityIndicator,
    Button,
    Alert,
    StatusBar,
    Dimensions
} from "react-native";
import IoniconsIcon from "react-native-vector-icons/Ionicons";
import EntypoIcon from "react-native-vector-icons/Entypo";
import {firebaseApp} from '../config/DatabaseConfig';
import styles from '../styles/LoginStyles.js';

const SCREEN_HEIGHT = Dimensions.get('window').height
const SCREEN_WIDTH = Dimensions.get('window').width

const LoginScreen = (props) => {
    //
    const [loading, setLoading] = useState(false);
    const startLoading = () => {
        setLoading(true);
        setTimeout(() => {
            setLoading(false);
        }, 3000);
    };
    const [email, setEmail] = useState(""); // email for login
    const [password, setPassword] = useState(""); // password for login
    // Function for login. Uses firebase authentication
    const loginFunc = () => {
      //activates the activity indicator's animation
      startLoading();
      // check if valid login
      firebaseApp
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(() => props.navigation.navigate('Main Screen'))
      .catch(error => Alert.alert(
          "Login",
          "Invalid login information",
          [
            { text: "OK", onPress: () => console.log("OK Pressed") }
          ],
          { cancelable: false }
        )
      )   
    };

    // Function for login. Uses firebase authentication
    const debugLoginFunc = () => {
        //activates the activity indicator's animation
        startLoading();
        // check if valid login
        firebaseApp
        .auth()
        .signInWithEmailAndPassword("test@radford.edu", "password")
        .then(() => props.navigation.navigate('Main Screen'))
        .catch(error => Alert.alert(
            "Login",
            "Invalid login information",
            [
              { text: "OK", onPress: () => console.log("OK Pressed") }
            ],
            { cancelable: false }
          )
        )   
      };

    // Function for registering an account. Uses firbase's authentication API to signup
    const registerFunc = () => {
        if(email === '' && password === '') {
            Alert.alert('Enter details to signup!')
          } 
        else {
            firebaseApp
            .auth()
            .createUserWithEmailAndPassword(email, password)
            .then(() => Alert.alert(
                "Register",
                "You're registered! Enjoy!",
                [
                    {text: "OK", onPress: () => console.log("Register OK pressed") }
                ],
                { cancelable: false }
            ))
            .catch(error => Alert.alert(
                "Register",
                "Invalid information",
                [
                    {text: "OK", onPress: () => console.log("OK pressed") }
                ],
                { cancelable: false }
            ))
        }
    };

    return (
        <View style={styles.container}>
            <StatusBar
                backgroundColor="#E35614"
                />
            <ImageBackground
                source={require("../assets/images/login_bg.png")}
                resizeMode="cover"
                style={styles.image}
                imageStyle={styles.image_imageStyle}
            >
                <Image
                    source={require("../assets/images/logo.png")}
                    resizeMode="contain"
                    style={styles.logo}
                ></Image>
                <Text style={styles.tenderRecipes}>Tender Recipes</Text>


                <View style={styles.usernameGroup}>
                    <IoniconsIcon name="md-person" style={styles.icon}></IoniconsIcon>
                    <TextInput
                        testID='emailInput'
                        placeholder="Email"
                        placeholderTextColor="rgba(230, 230, 230,1)"
                        inlineImagePadding={0}
                        selectionColor="rgba(52,250,215,1)"
                        style={styles.textInput}
                        onChangeText={(email) => setEmail(email)}
                    ></TextInput>
                </View>
                <View style={styles.passwordGroup}>
                    <EntypoIcon name="lock" style={styles.icon2}></EntypoIcon>
                    <TextInput
                        testID='passwordInput'
                        placeholder="Password"
                        placeholderTextColor="rgba(230, 230, 230,1)"
                        selectionColor="rgba(52,250,215,1)"
                        secureTextEntry={true}
                        style={styles.textInput2}
                        onChangeText={(password) => setPassword(password)}
                    ></TextInput>
                </View>


                <View style={styles.loginBtnRow}>
                    <TouchableOpacity
                        onPress={ loginFunc } //insert navigation
                        style={styles.loginBtn}
                        testID='loginButton'
                    >
                        <Text style={styles.login}>Login</Text>
                    </TouchableOpacity>
                    <TouchableOpacity
                        onPress={() => props.navigation.navigate('RegisterScreen')}
                        testID='registerButton'
                        //onPress={registerFunc} //insert navigation
                        style={styles.signupBtn}
                    >
                        <Text style={styles.signUp}>Sign up</Text>
                    </TouchableOpacity>
                    {/*DEBUG BUTTON*/}
                    <TouchableOpacity
                        onPress={ debugLoginFunc }
                        //onPress={registerFunc} //insert navigation
                        style={styles.signupBtn}
                    >
                        <Text style={styles.signUp}>Debug Login</Text>
                    </TouchableOpacity>
                </View>

                <ActivityIndicator
                    animating={loading}
                    color="rgba(254,242,94,1)"
                    size="large"
                    style={styles.activityIndicator}
                ></ActivityIndicator>

            </ImageBackground>
        </View>
    );
}

export default LoginScreen;