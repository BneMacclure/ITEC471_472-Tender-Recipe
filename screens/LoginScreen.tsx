import React, { Component } from "react";
import {
    StyleSheet,
    View,
    Image,
    ImageBackground,
    Text,
    TextInput,
    TouchableOpacity,
    ActivityIndicator
} from "react-native";
import IoniconsIcon from "react-native-vector-icons/Ionicons";
import EntypoIcon from "react-native-vector-icons/Entypo";

function LoginScreen(props) {
    return (
        <View style={styles.container}>
            <ImageBackground
                source={require("../assets/images/angryimg_(1)1.png")}
                resizeMode="cover"
                style={styles.image}
                imageStyle={styles.image_imageStyle}
            >
                <Image
                    source={require("../assets/images/output-onlinepngtools_(2)1.png")}
                    resizeMode="contain"
                    style={styles.image2}
                ></Image>
                <Text style={styles.tenderRecipes}>Tender Recipes</Text>
                <View style={styles.iconRow}>
                    <IoniconsIcon name="md-person" style={styles.icon}></IoniconsIcon>
                    <TextInput
                        placeholder=" Username"
                        placeholderTextColor="rgba(230, 230, 230,1)"
                        inlineImagePadding={0}
                        selectionColor="rgba(52,250,215,1)"
                        style={styles.textInput}
                    ></TextInput>
                </View>
                <View style={styles.icon2Row}>
                    <EntypoIcon name="lock" style={styles.icon2}></EntypoIcon>
                    <TextInput
                        placeholder=" Password"
                        placeholderTextColor="rgba(230, 230, 230,1)"
                        selectionColor="rgba(52,250,215,1)"
                        secureTextEntry={true}
                        style={styles.textInput2}
                    ></TextInput>
                </View>
                <View style={styles.loginStackRow}>
                    <View style={styles.loginStack}>
                        <Text style={styles.login}>Login</Text>
                        <TouchableOpacity style={styles.button}></TouchableOpacity>
                    </View>
                    <View style={styles.signUpStack}>
                        <Text style={styles.signUp}>Sign up</Text>
                        <TouchableOpacity style={styles.button1}></TouchableOpacity>
                    </View>
                </View>
                <ActivityIndicator
                    color="rgba(254,242,94,1)"
                    size="large"
                    style={styles.activityIndicator}
                ></ActivityIndicator>
            </ImageBackground>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1
    },
    image: {
        flex: 1
    },
    image_imageStyle: {},
    image2: {
        width: 200,
        height: 200,
        borderRadius: 57,
        borderWidth: 4,
        borderColor: "rgba(255,255,255,1)",
        borderStyle: "solid",
        marginTop: 57,
        marginLeft: 80
    },
    tenderRecipes: {
        fontFamily: "big-shoulders-display-700",
        color: "rgba(251,251,251,1)",
        fontSize: 36,
        marginTop: 17,
        marginLeft: 89
    },
    icon: {
        color: "rgba(255,255,255,1)",
        fontSize: 31,
        height: 34,
        width: 23,
        marginTop: 1
    },
    textInput: {
        fontFamily: "helvetica-regular",
        color: "rgba(253,253,253,1)",
        width: 276,
        borderRadius: 10,
        borderWidth: 3,
        borderColor: "rgba(255,255,255,1)",
        shadowColor: "rgba(206,203,203,1)",
        shadowOffset: {
            height: 3,
            width: 3
        },
        elevation: 5,
        shadowOpacity: 1,
        shadowRadius: 0,
        lineHeight: 14,
        letterSpacing: 1,
        textAlign: "left",
        inlineImageLeft: "0",
        fontSize: 16,
        height: 35,
        marginLeft: 4
    },
    iconRow: {
        height: 35,
        flexDirection: "row",
        marginTop: 69,
        marginLeft: 31,
        marginRight: 26
    },
    icon2: {
        color: "rgba(255,255,255,1)",
        fontSize: 31,
        height: 35,
        width: 31
    },
    textInput2: {
        fontFamily: "helvetica-regular",
        color: "#121212",
        height: 35,
        width: 276,
        borderRadius: 10,
        borderWidth: 3,
        borderColor: "rgba(255,255,255,1)",
        shadowColor: "rgba(206,203,203,1)",
        shadowOffset: {
            width: 3,
            height: 3
        },
        elevation: 5,
        shadowOpacity: 1,
        shadowRadius: 0,
        fontSize: 16
    },
    icon2Row: {
        height: 35,
        flexDirection: "row",
        marginTop: 26,
        marginLeft: 27,
        marginRight: 26
    },
    login: {
        top: 11,
        position: "absolute",
        fontFamily: "helvetica-regular",
        color: "rgba(249,71,35,1)",
        letterSpacing: 1,
        fontSize: 14,
        left: 55
    },
    button: {
        top: 0,
        position: "absolute",
        backgroundColor: "rgba(253,253,253,1)",
        borderRadius: 10,
        left: 0,
        height: 36,
        width: 150
    },
    loginStack: {
        width: 150,
        height: 36
    },
    signUp: {
        top: 11,
        position: "absolute",
        fontFamily: "helvetica-regular",
        color: "rgba(249,71,35,1)",
        letterSpacing: 1,
        fontSize: 14,
        left: 48
    },
    button1: {
        top: 0,
        position: "absolute",
        backgroundColor: "rgba(253,253,253,1)",
        borderRadius: 10,
        left: 0,
        height: 36,
        width: 150
    },
    signUpStack: {
        width: 150,
        height: 36,
        marginLeft: 6
    },
    loginStackRow: {
        height: 36,
        flexDirection: "row",
        marginTop: 30,
        marginLeft: 27,
        marginRight: 27
    },
    activityIndicator: {
        marginTop: 34,
        marginLeft: 161
    }
});

export default LoginScreen;
