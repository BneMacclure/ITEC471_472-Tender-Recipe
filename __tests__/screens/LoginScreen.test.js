import 'react-native';
import React, { Component , useState}  from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from '../../screens/LoginScreen';
import { Alert } from 'react-native';
import {act, fireEvent, waitFor, cleanup} from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
jest.mock('../../screens/LoginScreen');
import Enzyme, { configure, shallow, mount, render } from 'enzyme';

beforeEach(() => {
  // Clear all instances and calls to constructor and all methods:
  LoginScreen.mockClear();

});
describe('<LoginScreen />', () => {
    it('LoginScreen snapshot test' , () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
	it('first LoginScreen page renders',()=>{
		const wrapper = render(<LoginScreen />);
		expect(wrapper).toBeTruthy();
	});

	it('first LoginScreen page renders',()=>{
		const wrapper = render(<LoginScreen />);
		
		let log =  new LoginScreen();
		log.debugLoginFunc;
		expect(LoginScreen).toHaveBeenCalledTimes(2);
		
	});
	
	it("should update state on click", () => {
		window.alert = jest.fn();
		changeinput=jest.fn();
		const wrapper = mount(<LoginScreen onChange={changeinput} />);
		jest.spyOn(Wrapper.instance(), loginFunc);
		wrapper.find("loginButton").simulate("click");
		expect(formWrapper.instance().loginFunc).toBeCalled();
		expect(window.alert).toBeCalledWith(loginFunc);
	});
	
	
})

