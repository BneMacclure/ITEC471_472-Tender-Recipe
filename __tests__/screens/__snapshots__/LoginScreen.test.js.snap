// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> LoginScreen snapshot test 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {},
        ]
      }
    />
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "rgba(255,255,255,1)",
          "borderRadius": 57,
          "borderWidth": 4,
          "height": 200,
          "marginTop": 88.93333333333334,
          "width": 200,
        }
      }
    />
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "rgba(251,251,251,1)",
          "fontFamily": "BigShouldersDisplay_700Bold",
          "fontSize": 36,
          "marginTop": 10,
        }
      }
    >
      Tender Recipes
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "height": 35,
          "marginRight": 30,
          "marginTop": 60,
          "width": 276,
        }
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        inlineImagePadding={0}
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="rgba(230, 230, 230,1)"
        rejectResponderTermination={true}
        selectionColor="rgba(52,250,215,1)"
        style={
          Object {
            "borderColor": "rgba(255,255,255,1)",
            "borderRadius": 10,
            "borderWidth": 3,
            "color": "rgba(255,255,255,1)",
            "fontSize": 16,
            "height": 35,
            "letterSpacing": 1,
            "lineHeight": 14,
            "marginLeft": 4,
            "paddingLeft": 8,
            "textAlign": "left",
            "width": 275,
          }
        }
        testID="emailInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "height": 35,
          "marginRight": 30,
          "marginTop": 26,
          "width": 276,
        }
      }
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="rgba(230, 230, 230,1)"
        rejectResponderTermination={true}
        secureTextEntry={true}
        selectionColor="rgba(52,250,215,1)"
        style={
          Object {
            "borderColor": "rgba(255,255,255,1)",
            "borderRadius": 10,
            "borderWidth": 3,
            "color": "rgba(255,255,255,1)",
            "fontSize": 16,
            "height": 35,
            "letterSpacing": 1,
            "paddingLeft": 8,
            "width": 275,
          }
        }
        testID="passwordInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "elevation": 5,
          "flexDirection": "row",
          "height": 36,
          "marginTop": 30,
          "shadowColor": "rgba(206,203,203,1)",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 1,
          "shadowRadius": 0,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,1)",
            "borderRadius": 10,
            "height": 36,
            "justifyContent": "center",
            "opacity": 1,
            "width": 149,
          }
        }
        testID="loginButton"
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "rgba(249,71,35,1)",
            }
          }
        >
          Login
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,1)",
            "borderRadius": 10,
            "height": 36,
            "justifyContent": "center",
            "marginLeft": 4,
            "opacity": 1,
            "width": 149,
          }
        }
        testID="registerButton"
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "rgba(249,71,35,1)",
            }
          }
        >
          Sign up
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,1)",
            "borderRadius": 10,
            "height": 36,
            "justifyContent": "center",
            "marginLeft": 4,
            "opacity": 1,
            "width": 149,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "rgba(249,71,35,1)",
            }
          }
        >
          Debug Login
        </Text>
      </View>
    </View>
    <ActivityIndicator
      animating={false}
      color="rgba(254,242,94,1)"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "alignSelf": "center",
          "marginTop": 50,
        }
      }
    />
  </View>
</View>
`;
